nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -142,7 +186,7 @@ in
       wrappedSlurm = pkgs.stdenv.mkDerivation {
         name = "wrappedSlurm";
 
-        propagatedBuildInputs = [ cfg.package configFile ];
+        propagatedBuildInputs = [ cfg.package etcSlurm ];
```

Comment:
What is the purpose of propagated build inputs here? If it is merely to ensure that they end up in the runtime closure, referencing them in the wrapper is sufficient to achieve that.

Reply:
That is a very good question. The `wrappedSlurm` derivation only contains the wrappers (and now the man pages). Is the `propagatedBuildInputs = [ cfg.package ]` even necessary here?

Reply:
It depends on what `buildEnv` does with propagated build inputs, maybe it does work as intended, though I'd expect `propagatedUserEnvPkgs` to be more appropriate.

Reply:
`propagatedBuildInputs` does not propagate the paths into the new package. That's the reason why the man pages were missing. I am not sure how `propagatedUserEnvPkgs` would work here?

Reply:
But this is a `mkDerivation` and the builder is replaced by the custom builder script. Would `propagatedUserEnvPkgs` still work in that case?

Reply:
Well, no ... I hadn't noticed that it uses a custom builder ... I also see that it explicitly links in the manpage so propagation would be pointless anyway :)

Reply:
In other cases, `propagatedUserEnvPkgs` is the correct choice, however, but ofc. requires that the builder at least runs the standard fixup phase or something like it.

Reply:
Then the conclusion is that we can remove the line `propagatedBuildInputs = [ cfg.package configFile ];` entirely. I just ran the test without it. It still works.

Reply:
Absolutely, given that it doesn't run the fixup phase, it made no difference to begin with